### PR TITLE
validate stack name on edit

### PR DESCRIPTION
**Pull request by kinarashah**
[rancher/rancher#10208](https://github.com/rancher/rancher/issues/10208)

### CODE REVIEW

**Comment by alena1108** [comment]
@kinarashah need integration test for it ;)

**Comment by alena1108** [comment]
@kinarashah there is a NPE in the drone build

File: tests/integration/cattletest/core/test_stack.py [hunk 1]
```diff
@@ -25,3 +26,26 @@ def test_create_stack_binding(client):
...
+    with pytest.raises(ApiError) as e:
+        client.create_stack(name=env.name)
+    assert e.value.error.status == 422
+    assert e.value.error.code == 'NotUnique'
+    assert e.value.error.fieldName == 'name'
+    with pytest.raises(ApiError) as e:
+        client.update(env, name=env.name)
+    assert e.value.error.status == 422
+    assert e.value.error.code == 'NotUnique'
+    assert e.value.error.fieldName == 'name'
```
Comment (alena1108) [inline tests/integration/cattletest/core/test_stack.py]: need a new line

File: code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/StackValidationFilter.java [hunk 2]
```diff
@@ -81,4 +73,19 @@ private void validateInfraAccess(ApiRequest request, Stack stack, String action)
...
+        ResourceManager rm = locator.getResourceManagerByType(type);
+        Map<Object, Object> criteria = new HashMap<>();
+        criteria.put(ObjectMetaDataManager.NAME_FIELD, stackName);
+        criteria.put(ObjectMetaDataManager.REMOVED_FIELD, new Condition(ConditionType.NULL));
+        List<?> existingEnv = rm.list(type, criteria, null);
+        if (!existingEnv.isEmpty()) {
+                ValidationErrorCodes.throwValidationError(ValidationErrorCodes.NOT_UNIQUE, "name");
+        }
+    }
+}
```
Comment (alena1108) [inline code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/StackValidationFilter.java]: New line :)

**Comment by alena1108** [comment]
@kinarashah looks like a lot of tests are failing. Can you check them all, and run locally to see what needs to be corrected.

**Comment by kinarashah** [comment]
@alena1108 now skipping the duplicate check if name is same as previous stack name. v2 tests passing, one of the v1 tests is failing.

**Comment by alena1108** [comment]
LGTM

**Pull request merged**